source/distributed/tria.cc

```diff
@@ -4492,12 +4367,12 @@ namespace parallel
     {
       std::size_t mem=
         this->dealii::Triangulation<dim,spacedim>::memory_consumption()
-        + MemoryConsumption::memory_consumption(mpi_communicator)
-        + MemoryConsumption::memory_consumption(my_subdomain)
+        + MemoryConsumption::memory_consumption(this->mpi_communicator)
+        + MemoryConsumption::memory_consumption(this->my_subdomain)
```

Comment:
This is slightly awkward: these are now member variables of the (intermediate) base class. A cleaner approach would add a `memory_consumption` function to the intermediate base class and here simply add what that function reports plus the member variables specific to the current class.